UKHomeOffice/sgmr-service
# NMSW-1371 Upgrade to react router 6

## AC

Upgrade to react router 6
update app router
swap out history for navigate as per rrv6

This PR includes a few initial state fixes, most will be done in next PR (see list of broken pages for where this needs addressing)

----

## To test

- remove node modules
- run npm ci

THINGS THAT SHOULD WORK
- Footer links 
- create and activate a new account
- complete onboarding forms
- dashboard
- manage voyages
- create a new voyage
- new voyage departure
- new voyage arrival
- people page
- add a person
- remove a person
- edit a person
- pleasure craft page
- add a pleasure craft
- account
- edit account
- sign out

----

## Notes

Currently broken
- many unit tests
- check your answers page
- voyage flow from pleasure craft onwards
- Back link styling
- Edit pleasure craft
- Alert banner to confirm delete of person (assuming pc and voyage too)
- Resend verification email link
